components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/util/ActionValidator.java

```diff
@@ -21,24 +21,183 @@
+                validateForBlank(ActionMgtConstants.PASSWORD_FIELD,
+                        authentication.getProperty(Authentication.Property.PASSWORD).getValue());
+                break;
+            case BEARER:
+                validateForBlank(ActionMgtConstants.ACCESS_TOKEN_FIELD,
+                        authentication.getProperty(Authentication.Property.ACCESS_TOKEN).getValue());
+                break;
+            case API_KEY:
+                String apiKeyHeader = authentication.getProperty(Authentication.Property.HEADER).getValue();
+                validateForBlank(ActionMgtConstants.API_KEY_HEADER_FIELD, apiKeyHeader);
+                validateHeader(apiKeyHeader);
+                validateForBlank(ActionMgtConstants.API_KEY_VALUE_FIELD,
+                        authentication.getProperty(Authentication.Property.VALUE).getValue());
+                break;
+            case NONE:
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Validates the list of allowed headers for an action.
+     *
+     * @param allowedHeaders List of allowed headers to validate.
+     * @throws ActionMgtClientException If any header is invalid or excluded.
+     */
+    public void doValidateAllowedHeaders(List<String> allowedHeaders)
+            throws ActionMgtClientException {
+
+        if (allowedHeaders == null) {
+            return;
+        }
+
+        for (String header : allowedHeaders) {
+            validateForBlank(ActionMgtConstants.ALLOWED_HEADERS_FIELD, header);
+            validateHeader(header);
+        }
+        validateAllowedHeaders(allowedHeaders);
+    }
+
+    /**
+     * Validates the list of allowed parameters for an action.
+     *
+     * @param allowedParameters List of allowed parameters to validate.
+     * @throws ActionMgtClientException If any parameter is invalid or excluded.
+     */
+    public void doValidateAllowedParams(List<String> allowedParameters)
+            throws ActionMgtClientException {
+
+        if (allowedParameters == null) {
+            return;
+        }
+
+        for (String param : allowedParameters) {
+            validateForBlank(ActionMgtConstants.ALLOWED_PARAMETERS_FIELD, param);
+            validateParameter(param);
+        }
+        validateAllowedParameters(allowedParameters);
+    }
+
+    /**
+     * Validate configured headers by filtering out excluded headers that are configured at server level.
+     *
+     * @param allowedHeadersInAction List of allowed headers configured at action level.
+     * @throws ActionMgtClientException If any header is excluded by the server configuration.
+     */
+    private void validateAllowedHeaders(List<String> allowedHeadersInAction) throws ActionMgtClientException {
+
+        List<String> excludedHeadersServerConfig = ACTION_MGT_CONFIG.getPropertyValues(
+                ActionManagementConfig.ActionTypeConfig.PRE_ISSUE_ACCESS_TOKEN.getExcludedHeadersProperty());
```

Comment:
We can't scope this only for PRE_ISSUE_ACCESS_TOKEN action type, as there will be server wise excluded headers for other action types as well